api/w2v_helpers.py

```diff
@@ -170,28 +226,193 @@ def _remove_dislikes(bad_movies, good_movies_vec, rejected_list=[], harshness=1)
-            ids = [x[0] for x in recs]
-            return len(list(set(ids) & set(val_list)))
-
         aggregated = _aggregate_vectors(input, checked_list)
         recs = _similar_movies(aggregated, bad_movies, n=n)
         recs = _remove_dupes(recs, input, bad_movies, hist_list, checked_list + rejected_list)
         formatted_recs = [self._get_info(x[0], x[1]) for x in recs]
-        if val_list:
-            if return_scores:
-                return tuple([_score_model(recs, val_list), sum([i[3] for i in formatted_recs if i[3] is not None])/len(formatted_recs)])
-            elif scoring:
-                print(f"The model recommended {_score_model(recs, val_list)} movies that were on the watchlist!\n")
-                print(f"\t\t Average Rating: {sum([i[3] for i in formatted_recs if i[3] is not None])/len(formatted_recs)}\n")
-        if show_vibes:
-            print("You'll get along with people who like: \n")
-            for x in dupes:
-                print(self._get_info(x[0], x[1]))
-            print('\n')
-        if rec_movies:
-            return formatted_recs
+        return formatted_recs
+    
+    def get_recommendations(self, payload):
+        # request data 
+        user_id = payload.user_id
+        n = payload.number_of_recommendations
+        good_threshold = payload.good_threshold
+        bad_threshold = payload.bad_threshold
+        harshness = payload.harshness
+
+        """ create cursor """
+        try:
+            self.cursor_dog = self.get_cursor()
+            print("Connected!")
+        except Exception as e:
+            print("Connection problem chief!\n")
+            print(e)
+
+        """ Check if user has ratings data in IMDB or Letterboxd or Groa"""
+        # should be just one call to a single table called `ratings`
+        # if one table can just do a single call to get all rows with user_id=%s
+        # and then just check the len(ratings)
+        query = "SELECT EXISTS(SELECT 1 FROM user_letterboxd_ratings where user_id=%s);"
+        self.cursor_dog.execute(query, (user_id,))
+        boolean_letterboxd = self.cursor_dog.fetchall()
+        if boolean_letterboxd[0][0]==False:
+            print("user_id not found in Letterboxd ratings")
+
+        query = "SELECT EXISTS(SELECT 1 FROM user_imdb_ratings where user_id=%s);"
+        self.cursor_dog.execute(query, (user_id,))
+        boolean_imdb = self.cursor_dog.fetchall()
+        if boolean_imdb[0][0]==False:
+            print("user_id not found in IMDB ratings")
+
+        query = "SELECT EXISTS(SELECT 1 FROM user_groa_ratings where user_id=%s);"
+        self.cursor_dog.execute(query, (user_id,))
+        boolean_groa = self.cursor_dog.fetchall()
+        if boolean_groa[0][0]==False:
+            print("user_id not found in Groa ratings")
+
+        no_rating_data = False
+        if ((boolean_letterboxd[0][0]==False) & (boolean_imdb[0][0]==False) & (boolean_groa[0][0]==False)):
```

Comment:
& is the `bitwise and` operator, you want to use `and` here
```
>>> True & False
False
```

Reply:
Yeah that needs to return `False` there because if one is `True` than that means there is rating_data. And this will be removed as well once the db schema is adjusted and `reviews` is adjusted to be a single column